xonsh/history.py

```diff
@@ -115,14 +115,14 @@ def unlocked_files(self):
         for f in fs:
             try:
                 lj = lazyjson.LazyJSON(f, reopen=False)
-                if lj['locked']:
-                    continue
+                if only_unlocked:
+                    if lj['locked']:
+                        continue
                 # info: closing timestamp, number of commands, filename
                 files.append((lj['ts'][1], len(lj.sizes['cmds']) - 1, f))
                 lj.close()
             except (IOError, OSError, ValueError):
                 continue
-        files.sort()
```

Comment:
Ahh, I forgot about this. When you take into account locked files the "closing timestamp" is sometimes `None`, then the list can no longer be sorted, get an exception. For garbage collect the order of the files does not matter, does it?

I meant to sort the individual commands by their timestamps in both history.py files for PTK and readline, but I forgot. I'll push this with the other recommended changes sometime soon.

If you have objections please let me know so I can take them into account.